lib/spack/spack/util/compression.py

```diff
@@ -55,11 +55,50 @@ def tar_support():
         return False
 
 
+def tgz_support():
+    return tar_support() and gzip_support()
+
+
+def tbz_support():
+    return tar_support() and bz2_support()
+
+
+def txz_support():
+    return tar_support() and lzma_support()
+
+
 def allowed_archive(path):
     return False if not path else \
         any(path.endswith(t) for t in ALLOWED_ARCHIVE_TYPES)
 
 
+def lazy_caller(func_a, func_b, *callables):
```

Comment:
I don't see this used anywhere.

Reply:
It's not. I wrote this before I realized that 7zip can atomically decompress/extract .txz files and implemented the rename strategy. If the renaming is acceptable, then I'll remove this.

Reply:
The renaming works, but if you prefer to use this and not rename, then that works too.

Is there documentation on any of the options/behavior of 7zip (i.e. did the documentation suggest changing the name to force extraction/unarchiving to occur together)?

Reply:
I'd prefer to just do the rename, so I'll remove this. I'm not sure if it's actually in the 7zip documentation, I just happened to noticed the `tar.xz` test was failing where the `.txz` test was not. 

Edit: After some searching I was unable to find a reference to this in the docs, so it seems like this may be undocumented behavior.

Reply:
As far as which approach is better, I should clarify: if there is documentation for one of the two methods (i.e. 7zip docs mention performing a separate step to untar) then I think that should be preferred. If both/neither are documented, then renaming seems good (although it would be worth adding a comment that this approach isn't documented).